http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/AppDescriptorDAO.scala

```diff
@@ -0,0 +1,29 @@
+package org.broadinstitute.dsde.workbench.leonardo.dao
+
+import cats.mtl.Ask
+import org.broadinstitute.dsde.workbench.leonardo.ContainerImage
+import org.broadinstitute.dsde.workbench.leonardo.model.LeoException
+import org.broadinstitute.dsde.workbench.model.TraceId
+
+trait AppDescriptorDAO[F[_]] {
+  def getDescriptor(path: String)(implicit ev: Ask[F, TraceId]): F[AppDescriptor]
+}
+
+final case class AppDescriptor(name: String,
```

Comment:
Maybe add a comment that this matches the schema here: https://github.com/DataBiosphere/terra-app#app-schema

I expect the schema will evolve going forward, so we'll need to keep this in sync.